cartographer_ros/cartographer_ros/pbstream_to_ros_map_main.cc

```diff
@@ -42,21 +43,15 @@ namespace {
 void Run(const std::string& pbstream_filename, const std::string& map_filestem,
          const double resolution) {
   ::cartographer::io::ProtoStreamReader reader(pbstream_filename);
+  ::cartographer::io::ProtoStreamDeserializer deserializer(&reader);
 
-  ::cartographer::mapping::proto::PoseGraph pose_graph;
-  CHECK(reader.ReadProto(&pose_graph));
-  ::cartographer::mapping::proto::AllTrajectoryBuilderOptions
-      all_trajectory_builder_options;
-  CHECK(reader.ReadProto(&all_trajectory_builder_options));
+  const auto& pose_graph = deserializer.pose_graph();
 
   LOG(INFO) << "Loading submap slices from serialized data.";
   std::map<::cartographer::mapping::SubmapId, ::cartographer::io::SubmapSlice>
       submap_slices;
-  for (;;) {
-    ::cartographer::mapping::proto::SerializedData proto;
-    if (!reader.ReadProto(&proto)) {
-      break;
-    }
+  ::cartographer::mapping::proto::SerializedData proto;
+  while (deserializer.ReadNextSerializedData(&proto)) {
```

Comment:
Does this clear `proto` before writing? Otherwise `proto` must be local in the loop. Same in the other tools.

Reply:
It's a feature of proto3 oneof's: as soon as you set one, all others are cleared